src/code/modules/remediation.js

```diff
@@ -0,0 +1,273 @@
+export const GUIDE_REMEDIATION_REQUESTED = "Guide remediation requested";
+export const STARTED_REMEDIATION = "Started remediation";
+export const ENDED_REMEDIATION = "Ended remediation";
+
+import { GUIDE_MESSAGE_RECEIVED, GUIDE_ALERT_RECEIVED } from './notifications';
+import { showNotification } from '../actions';
+import actionTypes from '../action-types';
+import GeneticsUtils from '../utilities/genetics-utils';
+
+const initialState = false;
+
+export default function notifications(state = initialState, action) {
+  switch (action.type) {
+    case GUIDE_REMEDIATION_REQUESTED:
+      if (action.attribute && action.challengeType) {
+        return {
+          attribute: action.attribute,
+          practiceCriteria: action.practiceCriteria,
+          challengeType: action.challengeType
+        };
+      }
+      else
+        return initialState;
+    case ENDED_REMEDIATION:
+      return initialState;
+    // otherwise keep the remediation challenge
+    default:
+      return state;
+  }
+}
+
+/**
+ * action creators
+ */
+
+export function requestRemediation(data) {
+  return (dispatch, getState) => {
+    if (getState().isRemediation) {
+      return;
+    }
+
+    dispatch({
+      type: GUIDE_REMEDIATION_REQUESTED,
+      attribute: data.context.attribute,
+      practiceCriteria: data.context.practiceCriteria,
+      challengeType: data.context.challengeType
+    });
+
+    dispatch({
+      type: actionTypes.MODAL_DIALOG_SHOWN,
+      mouseShieldOnly: true
+    });
+
+    let dialog = {
+      message: "~ALERT.START_REMEDIATION",
+      closeButton: {
+        action: "startRemediation"
+      },
+      arrowAsCloseButton: true,
+      isInterrupt: true
+    };
+    dispatch(showNotification(dialog));
+  };
+}
+
+export function startRemediation() {
+  return {
+    type: STARTED_REMEDIATION
+  };
+}
+
+export function endRemediation() {
+  return {
+    type: ENDED_REMEDIATION
+  };
+}
+
+/**
+ * challenge-specific remediation authoring
```

Comment:
I think this is a tricky one.

For the most part, I think challenges are set up using "authored" (that is, variable) properties, that are translated by the code into a set of state properties. In the authoring document, we allow the authors the ability to change a lot of variables. In remediation, it's very similar, but we only allow the ITS to set three variables (trait, practiceCriteria/on/x, and challengeType). But in both cases there is an element of translation of the authored props to the state, which is hidden in the code. In remediation, the code just has to make a lot more decisions.

To the extent that it would be nice for the authors to be able to change *some* things about remediation (e.g. the color of the default drakes) that could definitely be in the authoring JSON. But it would still be a process of merging the ITS props, the authoring props, and the in-code assumptions.

In that regard, I would think that the first time an author wants to change something about the defaults, we should put what we can into authoring (I would assume it would just be the base alleles, at least at the start).

But I'm not certain I agree that the way the remediation sets state is redundant with the authoring state. It's just that the authors are required to make more of those decisions, so we get to copy them over from the JSON.